2. Web API now calls Microsoft Graph/TodoListClient/MainWindow.xaml.cs

```diff
@@ -141,9 +141,39 @@ private async Task GetTodoList(bool isAppStarting)
             }
             else
             {
-                string failureDescription = await response.Content.ReadAsStringAsync();
-                MessageBox.Show($"{response.ReasonPhrase}\n {failureDescription}", "An error occurred while getting /api/todolist", MessageBoxButton.OK);
-            }
+                if (response.StatusCode == System.Net.HttpStatusCode.Forbidden && response.Headers.WwwAuthenticate.Any())
```

Comment:
This is the way of letting the client application asking user for consent for the Web API.